docs/docs/genai/eval-monitor/scorers/llm-judge/migration-guide.mdx

```diff
@@ -0,0 +1,570 @@
+        Answer yes or no.
+        """,
+        model="openai:/gpt-4",
+    )
+
+    # Direct integration with MLflow GenAI evaluate
+    results = mlflow.genai.evaluate(
+        data=test_data,
+        predict_fn=my_predict_fn,
+        scorers=[correctness_judge, groundedness_judge],
+    )
+    ```
+
+  </TabItem>
+</Tabs>
+
+## Migrating `custom_prompt_judge`
+
+<Tabs>
+  <TabItem value="legacy" label="Legacy Approach" default>
+    ```python
+    from mlflow.genai.judges import custom_prompt_judge
+
+    formality_judge = custom_prompt_judge(
+        name="formality_judge",
+        prompt_template="""
+        Check if {{response}} is formal.
+
+        [[formal]]: The response is formal
+        [[informal]]: The response is informal
+        """,
+        model="openai:/gpt-4",
+    )
+
+    feedback = formality_judge(response="Dear Sir/Madam...")
+    ```
+
+  </TabItem>
+  <TabItem value="new" label="New Approach">
+    ```python
+    from mlflow.genai.judges import make_judge
+
+    formality_judge = make_judge(
+        name="formality_judge",
+        instructions="""
+        Check if the response {{response}} is formal or informal.
+
+        Answer "formal" if the response uses professional language and proper etiquette.
+        Answer "informal" if the response uses casual or conversational language.
+        """,
+        model="openai:/gpt-4",
+    )
+
+    feedback = formality_judge(outputs={"response": "Dear Sir/Madam..."})
+    ```
+
+  </TabItem>
+</Tabs>
+
+## Key Differences
+
+### 1. Unified Instruction Format
+
+The new API uses a single `instructions` parameter with template variables instead of separate parameters for different components:
+
+```python
+# Instead of multiple parameters...
+is_correct(request="...", response="...", expected_facts=[...])
+
+# Use template variables in instructions
+make_judge(instructions="Check {{response}} against {{expected_facts}}...")
+```
+
+### 2. Flexible Input Organization
+
+The new API organizes data into `inputs`, `outputs`, and `expectations`:
+
+- **`inputs`**: Data provided to your application (e.g., user queries)
+- **`outputs`**: Results produced by your application (e.g., generated responses)
+- **`expectations`**: Ground truth or expected outcomes for evaluation
+
+:::note[Handling Optional Parameters]
+
+The `make_judge` API uses simple template variable substitution. If you need conditional logic, include all variables in your instructions and pass empty strings or placeholder values for optional parameters:
+
+```python
+# Include all variables in instructions
+judge = make_judge(
+    name="my_judge",
+    instructions="Check {{response}}. Expected: {{expected_response}}",
+    model="openai:/gpt-4",
+)
+
+# Pass empty string for optional parameters
+judge(
+    outputs={"response": "The answer"},
+    expectations={"expected_response": ""},  # Empty if not available
+)
+```
+
+:::
+
+### 3. Better Reusability
```

Comment:
The old methods had the same reusability that is shown in the examples. I think here we can show that the result of `make_judge()` can be passed directly to `genai.evaluate()` and also be used as a registered scorer. If we already cover these two bits in other sections then we can also remove this section.